### PR TITLE
generate pas file support record function/procedure

### PR DESCRIPTION
**Pull request by chongchong2018**
{ TTestMsg0Record }

type
  TTestMsg0Record = record
{$IFDEF VER_XE}
  public
{$ENDIF}
    Field1 : LongInt;
    Field2 : Int64;
  {$IFDEF VER_XE}
    procedure Init;
    procedure Finalise;
    function  EncodeData(var Buf; const BufSize: Integer): Integer;
    function  EncodeValue(var Buf; const BufSize: Integer): Integer;
    function  EncodeField(var Buf; const BufSize: Integer; const FieldNum: Integer): Integer;
    function  DecodeValue(const Buf; const BufSize: Integer): Integer;
    procedure DecodeField(const Field: TpbProtoBufDecodeField);
  {$ENDIF}
  end;
  PTestMsg0Record = ^TTestMsg0Record;

{$IFNDEF VER_XE}
procedure TestMsg0RecordInit(var A: TTestMsg0Record);
procedure TestMsg0RecordFinalise(var A: TTestMsg0Record);
function  pbEncodeDataTestMsg0Record(var Buf; const BufSize: Integer; const A: TTestMsg0Record): Integer;
function  pbEncodeValueTestMsg0Record(var Buf; const BufSize: Integer; const A: TTestMsg0Record): Integer;
function  pbEncodeFieldTestMsg0Record(var Buf; const BufSize: Integer; const FieldNum: Integer; const A: TTestMsg0Record): Integer;
function  pbDecodeValueTestMsg0Record(const Buf; const BufSize: Integer; var Value: TTestMsg0Record): Integer;
procedure pbDecodeFieldTestMsg0Record(const Field: TpbProtoBufDecodeField; var Value: TTestMsg0Record);
{$ENDIF}

-------------------------------------------------------------------------------------------------------------------
  TTestMsg0Record = record
  public
    Field1 : LongInt;
    Field2 : Int64;
    procedure Init;
    procedure Finalise;
    function  EncodeData(var Buf; const BufSize: Integer): Integer;
    function  EncodeValue(var Buf; const BufSize: Integer): Integer;
    function  EncodeField(var Buf; const BufSize: Integer; const FieldNum: Integer): Integer;
    function  DecodeValue(const Buf; const BufSize: Integer): Integer;
    procedure DecodeField(const Field: TpbProtoBufDecodeField);
  end;
  PTestMsg0Record = ^TTestMsg0Record;

### CODE REVIEW

File: Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr [hunk 1]
```diff
@@ -62,29 +64,50 @@ begin
...
-      else
-      if StrEqualNoAsciiCase(S, '--help') then
-        begin
-          PrintHelp;
-          Halt(1);
-        end
+      if StrMatchLeft(S, '--pas_ver=', False) then
+      begin
+        Delete(S, 1, 10);
+        N := StrToIntDef(S, 0);
```
Comment (butlerdj) [inline Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: [dcc32 Error] ProtoBufCodeGen.dpr(74): E2003 Undeclared identifier: 'N'

File: Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr [hunk 2]
```diff
@@ -62,29 +64,50 @@ begin
...
+        if (N >= Integer(Low(TCodeGenSupportVersion))) and (N <= Integer(High(TCodeGenSupportVersion))) then
+          ParamSupportVersion := TCodeGenSupportVersion(N);
```
Comment (butlerdj) [inline Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: [dcc32 Error] ProtoBufCodeGen.dpr(76): E2003 Undeclared identifier: 'ParamSupportVersion'

**Comment by butlerdj** [comment]
Current version does not compile, see file comments.

File: Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr (hunk 2 above)
Comment (chongchong2018) [reply Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: Se i changed flcProtoBufProtoCodeGenPascal.pas

Comment (butlerdj) [reply Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: Did you make a new commit? Not seeing changes. The .dpr file needs to change to declare those new variables.

Comment (butlerdj) [reply Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: Variables not declared in .dpr file, e.g.

var N : Integer;

Comment (butlerdj) [reply Source/ProtocolBuffers/ProtoBufCodeGenApp/ProtoBufCodeGen.dpr]: Thanks!

**Pull request merged**